ultravox/tools/ds_tool/ds_tool.py

```diff
@@ -223,6 +226,127 @@ def __post_init__(self):
             self.upload_split = self.dataset_split
 
 
+class DatasetChunkProcessor:
+    args: DatasetToolArgs
+    cache_dir: str = ".cache/ds_tool/processed_datasets"
+    chunks_not_uploaded: List[Tuple[int, int]] = []
+    total_samples_processed: Dict[str, int] = defaultdict(int)
+
```

Comment:
The logic and consistency in the naming used here are unclear.

If `split_name` is always required, is `total_samples_processed` a dictionary with only a single key?

What's the difference between `not_uploaded` and `processed`?  What's the difference between `total_samples_processed` vs a simpler name `samples_processed`?

Reply:
total_samples_processed can be removed now that we have huggingface README integraiton.

Reply:
not_uploaded are chunks that failed to be processed, and could not be split further.